testbed/testbed/load_generator.go

```diff
@@ -258,12 +258,6 @@ func (ps *ProviderSender) generateTrace() error {
 			return fmt.Errorf("cannot send traces: %w", err)
 		}
 		ps.nonPermanentErrors.Add(uint64(traceData.SpanCount()))
-
```

Comment:
would it be possible to change this to still exit this function eventually after a certain number of retries? Otherwise, some cases like [the one testing the MemoryLimiter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c1e2df6b5a5b7397a168c8b46f72db26ec3dd3af/testbed/tests/trace_test.go#L196) will stall infinitely - In the linked example the collector at one point rejects any incoming data items due to the memory limit being exceeded (which is a non permanent error) and the test does not terminate

Reply:
That's a great suggestion.
I'll implement a timeout based mechanism. Thanks for your review!

Reply:
thanks for the adaptation! just tried it out again with the test case I mentioned earlier and it seems to work well

Reply:
Glad to know it worked. Thanks for your quick review!